apps/emqx_connector/test/emqx_connector_redis_SUITE.erl

```diff
@@ -0,0 +1,140 @@
+% %%--------------------------------------------------------------------
+% %% Copyright (c) 2020-2022 EMQ Technologies Co., Ltd. All Rights Reserved.
+% %%
+% %% Licensed under the Apache License, Version 2.0 (the "License");
+% %% you may not use this file except in compliance with the License.
+% %% You may obtain a copy of the License at
+% %% http://www.apache.org/licenses/LICENSE-2.0
+% %%
+% %% Unless required by applicable law or agreed to in writing, software
+% %% distributed under the License is distributed on an "AS IS" BASIS,
+% %% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+% %% See the License for the specific language governing permissions and
+% %% limitations under the License.
+% %%--------------------------------------------------------------------
+
+-module(emqx_connector_redis_SUITE).
+
+-compile(nowarn_export_all).
+-compile(export_all).
+
+-include_lib("eunit/include/eunit.hrl").
+-include_lib("emqx/include/emqx.hrl").
+-include_lib("stdlib/include/assert.hrl").
+
+-define(REDIS_HOST, "redis").
+-define(REDIS_PORT, 6379).
+-define(REDIS_RESOURCE_MOD, emqx_connector_redis).
+
+all() ->
+    emqx_common_test_helpers:all(?MODULE).
+
+groups() ->
+    [].
+
+init_per_suite(Config) ->
+    case emqx_common_test_helpers:is_tcp_server_available(?REDIS_HOST, ?REDIS_PORT) of
+        true ->
+            ok = emqx_common_test_helpers:start_apps([emqx_conf]),
+            ok = emqx_connector_test_helpers:start_apps([emqx_resource, emqx_connector]),
+            Config;
+        false ->
+            {skip, no_redis}
+    end.
+
+end_per_suite(_Config) ->
+    ok = emqx_common_test_helpers:stop_apps([emqx_resource, emqx_connector]).
+
+init_per_testcase(_, Config) ->
+    Config.
+
+end_per_testcase(_, _Config) ->
+    ok.
+
+% %%------------------------------------------------------------------------------
+% %% Testcases
+% %%------------------------------------------------------------------------------
+
+t_single_lifecycle(_Config) ->
+    perform_lifecycle_check(
+        <<"emqx_connector_redis_SUITE_single">>,
+        redis_config_single(),
+        [<<"PING">>],
+        <<"PONG">>
+    ).
+
+t_cluster_lifecycle(_Config) ->
+    perform_lifecycle_check(
+        <<"emqx_connector_redis_SUITE_cluster">>,
+        redis_config_cluster(),
+        ["GET", "badkey"],
```

Comment:
Q: should this be a `PING/PONG` test like the others?

Reply:
Nope. Can't execute that against the cluster with the methods we call. That's why I even added the ability to pass in different commands and expected results: https://github.com/adrienmo/eredis_cluster/blob/master/src/eredis_cluster.erl#L412

Basically there is no key to use to do any sort of client side hashing to figure out which server to use for the ping. I think we might be able to if we used `eredis_cluster:qa/1` but that is not the method we use when querying in the connector module: https://github.com/emqx/emqx/blob/master/apps/emqx_connector/src/emqx_connector_redis.erl#L141